tritium-lib/src/main/java/com/palantir/tritium/proxy/InvocationEventProxy.java

```diff
@@ -87,13 +88,19 @@ public String toString() {
     private boolean isEnabled(Object instance, Method method, Object[] args) {
         try {
             return eventHandler.isEnabled()
-                    && filter.shouldInstrument(instance, method, args);
+                    && filter.shouldInstrument(instance, method, args)
+                    && (eventHandler.asyncSupport() || !isAsync(method));
```

Comment:
I think we want to remove this line. Tracing doesn't support async operations, but the tracing thread state is passed to executors, which will provide valuable information.